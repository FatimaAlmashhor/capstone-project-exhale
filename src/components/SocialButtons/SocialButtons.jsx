import React, { useState, useEffect } from 'react';
import firebase from 'firebase';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import { uiConfig } from '../../firebase';

const SocialButtons = () => {
  const [isSignedIn, setisSignedIn] = useState(false);
  useEffect(() => {
    firebase.auth().onAuthStateChanged((user) => {
      setisSignedIn(!!user);
    });
  });
  return (
    <div>
      {isSignedIn ? (
        <span>
          <div className="text-red-600">Signed In!</div>
          <button type="button" onClick={() => firebase.auth().signOut()}>
            Sign out!
          </button>
        </span>
      ) : (
        <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
      )}
    </div>
  );
};

export default SocialButtons;
