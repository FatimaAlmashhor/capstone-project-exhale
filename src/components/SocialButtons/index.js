import { faRoad } from "@fortawesome/free-solid-svg-icons"

export {default} from './SocialButtons'