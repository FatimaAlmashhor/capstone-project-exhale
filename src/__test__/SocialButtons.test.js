import React from 'react';
import renderer from 'react-test-renderer';
import SocialButtons from '../components/SocialButtons';

it('renders correctly', () => {
  const tree = renderer
    .create(<SocialButtons /> )
    .toJSON();
  expect(tree).toMatchSnapshot();
});