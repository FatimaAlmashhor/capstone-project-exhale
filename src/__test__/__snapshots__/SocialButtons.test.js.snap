// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    id="firebaseui_container"
  />
</div>
`;
